tests/ui/unsafe-binders/mismatch.rs

```diff
@@ -0,0 +1,33 @@
+#![feature(unsafe_binders)]
+//~^ WARN the feature `unsafe_binders` is incomplete
+
+use std::unsafe_binder::{wrap_binder, unwrap_binder};
+
+fn a() {
+    let _: unsafe<'a> &'a i32 = wrap_binder!(&());
```

Comment:
Also test

```rust
    let x = wrap_binder!(&42);
    let _: unsafe<'a> &'a i32 = x;
```

Reply:
And `let _: unsafe<'a> &'a u32 = wrap_binder!(&42);` to test inference into the binder's type

Reply:
The former should error with a type annotations needed error. 

The latter will work, but it's basically just the same thing as `tests/ui/unsafe-binders/expr.rs`.